std/math.d

```diff
@@ -4536,24 +4585,46 @@ if (is(typeof(rfunc(F.init)) : F) && isFloatingPoint!F)
  *
  * Unlike the rint functions, nearbyint does not raise the
  * FE_INEXACT exception.
+ *
+ * Note:
+ *     Not implemented for Microsoft C Runtime
  */
-real nearbyint(real x) @trusted nothrow @nogc
+real nearbyint(real x) @safe pure nothrow @nogc
 {
     version (CRuntime_Microsoft)
     {
-        assert(0);      // not implemented in C library
+        static assert(0, "nearbyintl not implemented in Microsoft C library");
     }
     else
         return core.stdc.math.nearbyintl(x);
 }
 
+///
+@safe pure unittest
+{
+    version (CRuntime_Microsoft) {}
+    else
+    {
+        assert(nearbyint(0.4) == 0);
+        assert(nearbyint(0.5) == 0);
+        assert(nearbyint(0.6) == 1);
+        assert(nearbyint(100.0) == 100);
+
+        assert(isNaN(nearbyint(real.nan)));
+        assert(nearbyint(real.infinity) == real.infinity);
+        assert(nearbyint(-real.infinity) == -real.infinity);
+    }
+}
+
 /**********************************
  * Rounds x to the nearest integer value, using the current rounding
  * mode.
+ *
  * If the return value is not equal to x, the FE_INEXACT
  * exception is raised.
- * $(B nearbyint) performs
- * the same operation, but does not set the FE_INEXACT exception.
+ *
+ * $(LREF nearbyint) performs the same operation, but does
+ * not set the FE_INEXACT exception.
  */
 real rint(real x) @safe pure nothrow @nogc { pragma(inline, true); return core.math.rint(x); }
```

Comment:
So we have `lround`, `nearbyint`, `round` and `rint` which all perform almost the same function. That's not very "abstracted high-level" over C  :/